### ISSUE TITLE
Redundant Curly Braces, Trailing Comma, and Auto Correct Problem

### ISSUE THREAD

**Issue by snowblink** [bbatsov/rubocop#2981]
It appears that the auto-correct removes the redundant curly braces, but still thinks it needs a comma, so we end up with two.

## Original code
```
          expect(blah).to eq({
            @abc => 0,
            @xyz => 1
          })
```

## Expected behavior
```
          expect(blah).to eq(
            @abc => 0,
            @xyz => 1,
          )
```

## Actual behavior
```
          expect(blah).to eq(
            @free_service => 0,
            @express_24_service => 17.96,,
          )
```

## Steps to reproduce the problem

`rubocop -Ra spec_file.rb`

## RuboCop version

Tested against latest tag and master
```
0.38.0 (using Parser 2.3.0.6, running on ruby 2.3.0 x86_64-darwin15)
```

**Comment by mikegee** [comment]
I failed to confirm on master. I got:

```ruby
expect(blah).to eq(@abc => 0,
                   @xyz => 1)
```

I put your original code in a `test.rb` file in rubocop's working directory, then ran `bin/rubocop -Ra test.rb`

Perhaps there's a difference in your `.rubocop.yml` that would explain this?

**Comment by snowblink** [comment]
```
Style/TrailingCommaInArguments:
  EnforcedStyleForMultiline: comma

Style/TrailingCommaInLiteral:
  EnforcedStyleForMultiline: comma
```

**Comment by Drenmi** [comment]
What happens is this:

1. `Style/TrailingCommaInArguments` comes along and inserts a comma:

```
expect(blah).to eq({
  @abc => 0,
  @xyz => 1
},)
```

2. `Style/TrailingCommaInLiteral` comes and inserts its own:

```
expect(blah).to eq({
  @abc => 0,
  @xyz => 1,
},)
```

3. `Style/BracesAroundHashParameters` removes the braces:

```
expect(blah).to eq(
  @abc => 0,
  @xyz => 1,,
)
```

I'll leave it for someone else to suggest how to solve it, while I get some sleep. 😄

**Comment by segiddins** [comment]
`Style/BracesAroundHashParameters` should probably also trim off the comma after the hash brace

**Comment by Drenmi** [comment]
After sleeping on it, I am thinking along the same lines @segiddins. At first it seemed strange this cop would check for it, but it is perfectly possible to have this to start with even without the cascading corrections:

```
expect(blah).to eq({
  @abc => 0,
  @xyz => 1,
},)
```
